### PR TITLE
Danielp iteration2 feature

### PR DESCRIPTION
**Pull request by danielnetwork**

### CODE REVIEW

File: Main.java [hunk 1]
```diff
@@ -3,221 +3,221 @@
...
+            }
+
+            menu = getMenu();
+
+            boolean exitProgram = false;
+
+            while (!exitProgram) {
+
+                output(
+                        "\n" + dateFormat.format(calendar.getDate())
```
Comment (DannyFGitHub) [inline Main.java]: Method call within a method call.

File: Main.java [hunk 2]
```diff
@@ -3,221 +3,221 @@
...
+            }
+        } catch (RuntimeException exception) {
+            output(exception);
+        }
+        output("\nEnded\n");
+    }
+
+
+    private static void payFine() {
+        new PayFineUI(new PayFineControl()).runState();
```
Comment (DannyFGitHub) [inline Main.java]: Issue 2. Method call within a method call.

File: Main.java [hunk 3]
```diff
@@ -3,221 +3,221 @@
...
+    private static void getMembers() {
+        output("");
+        for (Member member : library.getMembers()) {
+            output(member + "\n");
+        }
+    }
+
+
+    private static void borrowBook() {
+        new BorrowBookUI(new BorrowBookControl()).run();
```
Comment (DannyFGitHub) [inline Main.java]: Issue 3. Guideline 9 Method call within Method call

We could do:
private static void borrowBook() {
BorrowBookControl borrowBookControl = new BorrowBookControl();
new BorrowBookUI(borrowBookControl).run();

File: Main.java [hunk 4]
```diff
@@ -3,221 +3,221 @@
...
+    }
+
+
+    private static void returnBook() {
+        new ReturnBookUI(new ReturnBookControl()).run();
```
Comment (DannyFGitHub) [inline Main.java]: Issue 4. Guideline 9 Method call within Method call

We could do:
private static void returnBook() {
ReturnBookControl ReturnBookControl = new ReturnBookControl();
new ReturnBookUI(ReturnBookControl).run();

File: Main.java [hunk 5]
```diff
@@ -3,221 +3,221 @@
...
+    }
+
+
+    private static void fixBooks() {
+        new FixBookUI(new FixBookControl()).runState();
```
Comment (DannyFGitHub) [inline Main.java]: Issue 5. Guideline 9 Method call within Method call

We could do:
private static void fixBooks() {
FixBookControl fixBookControl = new FixBookControl();
new FixBookUI(fixBookControl).run();

File: Main.java [hunk 6]
```diff
@@ -3,221 +3,221 @@
...
+    }
+
+
+    private static void incrementDate() {
+        try {
+            int days = Integer.valueOf(input("Enter number of days: ")).intValue();
```
Comment (DannyFGitHub) [inline Main.java]: Issue 6. Guideline 9 Method call within Method call

We could do:
try {
String input = input("Enter number of days: ");
int days = Integer.valueOf(input).intValue();

File: Main.java [hunk 7]
```diff
@@ -3,221 +3,221 @@
...
+            calendar.incrementDate(days);
+            library.checkCurrentLoans();
+            output(dateFormat.format(calendar.getDate()));
```
Comment (DannyFGitHub) [inline Main.java]: Issue 7. Guideline 9 Method call within Method call

We could do:
library.checkCurrentLoans();
Date date = calendar.getDate();
StringBuffer formattedDate = dateFormat.format(date);
output(formattedDate);

File: Main.java [hunk 8]
```diff
@@ -3,221 +3,221 @@
...
+
+    }
+
+
+    private static void addMember() {
+        try {
+            String lastName = input("Enter last name: ");
+            String firstName = input("Enter first name: ");
+            String email = input("Enter email: ");
+            int phoneNumber = Integer.valueOf(input("Enter phone number: ")).intValue();
```
Comment (DannyFGitHub) [inline Main.java]: Issue 7. Guideline 9 Method call within Method call

We could do:
String email = input("Enter email: ");
String input = input("Enter phone number: ");
int phoneNumber = Integer.valueOf(input).intValue();

File: Loan.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("Loan:  ").append(ID).append("\n")
+                .append("  Borrower ").append(member.getID()).append(" : ")
+                .append(member.getLastName()).append(", ").append(member.getFirstName()).append("\n")
+                .append("  Book ").append(book.getID()).append(" : " )
+                .append(book.getTitle()).append("\n")
+                .append("  DueDate: ").append(dateFormat.format(date)).append("\n")
+                .append("  State: ").append(state);
```
Comment (DannyFGitHub) [inline Loan.java]: Issue 8. Guideline 9 Method call within Method call

We could do (this snippet below would start at line 53):
String memberID = member.getID();
String memberLastName = member.getLastName();
String memberFirstName = member.getFirstName();
String bookID = book.getID();
String bookTitle = book.getTitle();
StringBuffer formattedDate = dateFormat.format(date);

StringBuilder sb = new StringBuilder();
         sb.append("Loan:  ").append(ID).append("\n")
                 .append("  Borrower ").append(memberID).append(" : ")
                 .append(memberLastName).append(", ").append(memberFirstName).append("\n")
                 .append("  Book ").append(bookID).append(" : " )
                 .append(bookTitle).append("\n")
                 .append("  DueDate: ").append(formattedDate).append("\n")
                 .append("  State: ").append(state);

File: Main.java (hunk 1 above)
Comment (danielnetwork) [reply Main.java]: Resolved in commit ID: 26ba2da06855857a48bf0e9918244887fd37fe64

File: Main.java (hunk 2 above)
Comment (danielnetwork) [reply Main.java]: Resolved in commit ID: e847cf5b944e20f4a3bdf6016b5d857309ee12ca

File: Main.java (hunk 3 above)
Comment (danielnetwork) [reply Main.java]: Resolved in Commit ID: 6464f2c4dd344b6bc8af5e8ede7813d8bac25efe

File: Main.java (hunk 4 above)
Comment (danielnetwork) [reply Main.java]: Resolved in commit ID: d13ee5a3de37920b8924622a1b2c1c66302c2204

File: Main.java (hunk 5 above)
Comment (danielnetwork) [reply Main.java]: Resolved in commit ID: 17730e6c171d93258ff19d70f12305d9c5920ca1

File: Main.java (hunk 6 above)
Comment (danielnetwork) [reply Main.java]: Resolved in commit ID: 6505dfce9ae0291988ea2e277c017a75a1861841

File: Main.java (hunk 7 above)
Comment (danielnetwork) [reply Main.java]: Resolved in Commit ID: 8c0415d2f88fa53679f93b2dbea243646a9461ed

File: Main.java (hunk 8 above)
Comment (danielnetwork) [reply Main.java]: Resolved in Commit ID: 05986f5181186d3ae3a5133634e6cbd1081b7ad3

File: Loan.java (hunk 9 above)
Comment (danielnetwork) [reply Loan.java]: Resolved in Commit ID: b2f78a2494df33493f7177992d93ec5548b4ec3c

**Comment by danielnetwork** [comment]
All reported issues have been resolved

**Pull request merged**